core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -533,6 +534,25 @@ private static Schema applyChanges(
       }
     }
 
+    if (base != null) {
+      for (int fieldIdToDelete : deletes) {
+        base.specsById()
+            .forEach(
+                (specId, spec) -> {
+                  // Prevent dropping fields that are referenced in older partitions specs in use.
+                  if (!specId.equals(base.defaultSpecId())) {
```

Comment:
Wondering if we will run into the issue again if we allow dropping from current spec.

Reply:
This makes sense. My concern is that this is a big behavior change. Users cannot drop partition columns at all.